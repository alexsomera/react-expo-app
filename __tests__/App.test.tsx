import React from 'react';
import renderer from 'react-test-renderer';
import Index from '../app/index';

// Mocks necessÃ¡rios para o teste do componente Index
jest.mock('expo-router', () => ({
  useRouter: jest.fn(() => ({
    push: jest.fn(),
  })),
  useLocalSearchParams: jest.fn(() => ({})),
}));

jest.mock('expo-font', () => ({
  useFonts: jest.fn(() => [true, null]),
  loadAsync: jest.fn(),
}));

jest.mock('@expo/vector-icons', () => ({
  MaterialIcons: 'MaterialIcons',
}));

jest.mock('expo-constants', () => ({
  manifest: {
    extra: {
      eas: {
        projectId: 'your-project-id',
      },
    },
  },
}));

jest.mock('../components/SearchBar', () => 'SearchBar');

jest.mock('../ViewModel/UsersViewModel', () => ({
  useUsersViewModel: jest.fn(() => ({
    filteredUsers: [],
    searchQuery: '',
    setSearchQuery: jest.fn(),
  })),
}));

describe('App', () => {
  it('renders correctly', () => {
    const tree = renderer.create(<Index />).toJSON();
    expect(tree).toMatchSnapshot();
  });
});
