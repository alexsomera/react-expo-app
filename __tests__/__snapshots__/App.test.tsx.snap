// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`App renders correctly 1`] = `
<RCTScrollView
  style={
    {
      "backgroundColor": "#fff",
      "padding": 10,
      "paddingBlock": 30,
    }
  }
>
  <View>
    <Text
      style={
        {
          "fontSize": 22,
          "marginBottom": 20,
        }
      }
    >
      Selecione um usuário
    </Text>
    <SearchBar
      placeholder="Pesquisar usuários..."
      searchQuery=""
      setSearchQuery={[MockFunction]}
    />
  </View>
</RCTScrollView>
`;
